lib/smart_answer_flows/maternity-paternity-calculator/maternity_calculator_flow.rb

```diff
@@ -22,43 +23,23 @@ def define
           from { 2.years.ago(Time.zone.today) }
           to { 2.years.since(Time.zone.today) }
 
-          precalculate :leave_earliest_start_date do
-            calculator.leave_earliest_start_date
-          end
-
-          calculate :leave_start_date do |response|
-            ls_date = response
-            raise SmartAnswer::InvalidResponse if ls_date < leave_earliest_start_date
-
-            calculator.leave_start_date = ls_date
-            calculator.leave_start_date
+          on_response do |response|
+            self.leave_start_date = response
+            calculator.leave_start_date = leave_start_date
+            self.leave_end_date = calculator.leave_end_date
+            self.leave_earliest_start_date = calculator.leave_earliest_start_date
```

Comment:
```suggestion
```
We've still got the extra assignment here we don't need and can be removed. We don't need to assign` self.leave_earliest_start_date` at all, because we are using `calculator.leave_earliest_start_date` in the erb templates.